apps/hash-ai-worker-py/app/typesystem/data_type.py

```diff
@@ -18,3 +60,82 @@ class DataTypeSchema(OntologyTypeSchema, extra=Extra.allow):
     """
 
     ty: str = Field(..., alias="type")
+
+    def _type(self) -> type[DataType]:
+        match self.ty:
+            case "string":
+                return str
+            case "number":
+                return float
+            case "boolean":
+                return bool
+            case "null":
+                return type(None)
+            case "array":
+                return list[Any]
+            case "object":
+                return dict[str, Any]
+            case _:
+                # TODO: activities should never raise an exception, but instead return
+                #  an error value that the workflow can handle.
+                #  https://app.asana.com/0/0/1204934059777411/f
+                msg = f"Unknown type: {self.ty}"
+                raise ValueError(msg)
```

Comment:
isn't https://docs.python.org/3/library/typing.html#typing.assert_never more appropriate here?

Reply:
Yes, good point, thank you. Will address it when the PR is the top of my PR stack 🙂